### PR TITLE
Main loop plugins

### PR DESCRIPTION
**Pull request by oliver-sanders**
Companion to https://github.com/cylc/cylc-flow/pull/3492

Document the newfangled main-loop plugins.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
I think from a similar Simmental by Tim on the Sphinx-Cylc-extensions repo, that the Sphinx autosummary plugin is not simpatico with singlehtml, at lest when autobuilding. Simple solution, don’t build singlehtml, perhaps unhelpful. Will look and see if anyone has raised this with sphinxdoc at some point.

**Comment by oliver-sanders** [comment]
In the above there are phone typos for the words, “comment” and “compatible” that are so brilliant that I’m not going to edit the comment to fix them.

**Comment by kinow** [comment]
Links working on the `make clean html` version. Thanks @oliver-sanders !

File: src/user-guide/running-suites.rst [hunk 1]
```diff
@@ -1515,66 +1515,7 @@ and optionally, restart them on a different host.
...
-See the ``[suite servers]`` configuration section
-(:ref:`global-suite-servers`) for more details.
-
-
-.. [3] Late notification of clock-triggered tasks is not very useful in
-       any case because they typically do not depend on other tasks, and as
-       such they can often trigger on time even if the suite is delayed to
-       the point that downstream tasks are late due to their dependence on
-       previous-cycle tasks that are delayed.
+See :py:mod:`cylc.flow.main_loop.auto_restart` for details.
```
[the remark this answers is not in the document]
Comment (dwsutherland) [reply src/user-guide/running-suites.rst:1518]: Yes I found the same..

[the remark this answers is not in the document]
Comment (oliver-sanders) [reply src/user-guide/running-suites.rst:1518]: Ah, yeah, so these `:py:` thinggies are code references which only work if the code being referenced is auto-documented. The documentation system uses Python introspection so the thing you're documenting must be installed.

**Comment by oliver-sanders** [comment]
> And the "Next/Previous" navigation doesn't cycle through all the built-in plugins

Replied on Riot with a video on my branch.

**Comment by oliver-sanders** [comment]
Whilst we're all here, how do you feel about this approach of burying large quantities of documentation in the Cylc Flow codebase.

When you're documenting functions, classes, etc it makes good sense. When you are writing more user-guide oriented narrative, however, perhaps not quite perfect.

**Comment by kinow** [comment]
> Whilst we're all here, how do you feel about this approach of burying large quantities of documentation in the Cylc Flow codebase.
> 
> When you're documenting functions, classes, etc it makes good sense. When you are writing more user-guide oriented narrative, however, perhaps not quite perfect.

Fine either way I think. Only possible disadvantage I can think of, is if a user finds an error in the docs. That would require fixing in Cylc Flow, not in cylc-doc if that's auto-generated. Then probably the CI would take care to re-publish latest docs with the fix. For a user contributing, I think it's harder to preview the changes. But it could be easier to maintain for developers this way (i.e. code and docs are synced).

**Comment by oliver-sanders** [comment]
A couple of disadvantages from me:

* Hard to `git grep` for phrases.
* It is possible/easy to break the docs build due to simple errors in docstrings.

**Comment by kinow** [comment]
>It is possible/easy to break the docs build due to simple errors in docstrings.

At the same time forces developers to fix the doc in their code, a good side-effect I think.

**Comment by kinow** [comment]
Two approvals, LGTM

**Pull request merged**